### PR TITLE
Fix boundRatio incorrect merge

### PR DESCRIPTION
**Pull request by wangtZJU**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix boundRatio incorrect merge

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Missing tests. Missing comments.

**Comment by wangtZJU** [comment]
Hi @alexey-milovidov , the problem is that when `other.empty` = true, the original code will still use the `other.left.x`(which may be any arbitrary value) to update self's state.

I can reproduce it in a 2-nodes cluster,
```sql
-- this case mimic CI test 00715_bounding_ratio 
--
-- in node 1, run below queries
create table rate_test on cluster my_cluster (timestamp UInt32, event UInt32) engine=Memory;
insert into rate_test values (0,1000),(1,1001),(2,1002),(3,1003),(4,1004),(5,1005),(6,1006),(7,1007),(8,1008);
create table rate_test_dist (timestamp UInt32, event UInt32) engine=Distributed(my_cluster, currentDatabase(), 'rate_test');

select boundingRatio(timestamp, event) from rate_test;  -- output: 1
select boundingRatio(timestamp, event) from rate_test_dist; -- output: 126
```
but i don't know how to implement this in CI

**Comment by UnamedRus** [comment]
Probably, even more "interesting" output
https://fiddle.clickhouse.com/f42759d4-6cbd-4a3d-9261-da6fe6e654a7

**Comment by alexey-milovidov** [comment]
@wangtZJU, Write tests. Write comments. Write changelog entry.

**Comment by wangtZJU** [comment]
tests and changelog entry are added

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Thank you!

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 67caa733b1b6a903791cebfb86097c53e8a2c6c2 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60532/67caa733b1b6a903791cebfb86097c53e8a2c6c2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/AggregateFunctions/AggregateFunctionBoundingRatio.cpp [hunk 1]
```diff
@@ -63,6 +63,9 @@ struct AggregateFunctionBoundingRatioData
         {
             *this = other;
         }
+        else if (other.empty)
+        {
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/AggregateFunctionBoundingRatio.cpp:67]: It still lacks comments.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

File: tests/queries/0_stateless/00715_bounding_ratio.sql [hunk 2]
```diff
@@ -1,10 +1,19 @@
 drop table if exists rate_test;
+drop table if exists rate_test2;
 
 create table rate_test (timestamp UInt32, event UInt32) engine=Memory;
 insert into rate_test values (0,1000),(1,1001),(2,1002),(3,1003),(4,1004),(5,1005),(6,1006),(7,1007),(8,1008);
 
 select 1.0 = boundingRatio(timestamp, event) from rate_test;
 
+create table rate_test2 (timestamp UInt32, event UInt32) engine=Memory;
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/00715_bounding_ratio.sql:9]: It should be added as a new test. The old test should remain untouched.

**Comment by alexey-milovidov** [comment]
@wangtZJU,
> tests and changelog entry are added

But we also need comments in the code.
Writing code without comments is a crime.

**Comment by wangtZJU** [comment]
@alexey-milovidov fixed

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**